File: app/src/main/java/hexlet/code/games/PrimeGame.java
package hexlet.code.games;

import java.util.Random;

public final class PrimeGame extends AbstractGame {

    private static final int MAX_NUMBER = 47;

    public String getRule() {
        return "Answer 'yes' if given number is prime. Otherwise answer 'no'.";
    }

    public GameRound getNewRound() {
        var rand = new Random();
        var number = rand.nextInt(1, MAX_NUMBER + 1);
        var task = Integer.toString(number);

        return new GameRound(
                task,
                prepareAnswer(number)
        );
    }

    private String prepareAnswer(int number) {
        return isPrime(number) ? "yes" : "no";
    }

    public static boolean isPrime(int number) {
        if (number < 2) {
            return false;
        }

        var sqrt = Math.sqrt(number);

        for (var i = 2; i <= sqrt; i++) {
            if (number % i == 0) {
                return false;
            }
        }

        return true;
    }
}

File: app/src/main/kotlin/hexlet/code/PrimeGame.kt
package hexlet.code

class PrimeGame : Game() {
    override fun rule(): String {
        return "Answer 'yes' if given number is prime. Otherwise answer 'no'."
    }

    override fun newRound(): GameRound {
        val number = (1..numberLimit).random()

        val question = question(number)
        val rightAnswer = rightAnswer(number)

        return GameRound(question, rightAnswer)
    }

    private fun question(number: Int): String {
        return "Question: $number"
    }

    private fun rightAnswer(number: Int): String {
        return if (isPrime(number)) "yes" else "no"
    }

    private fun isPrime(number: Int): Boolean {
        val max = Math.sqrt(number.toDouble()).toInt()

        for (i in 2..max) {
            if (number % i == 0) return false
        }

        return true
    }

    companion object {
        private const val numberLimit = 47
    }
}
